contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -657,12 +684,29 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     // TRANSMIT_MSGDATA_CONSTANT_LENGTH_COMPONENT need to be changed accordingly
 
     uint256 initialGas = gasleft(); // This line must come first
+    Report memory report_ = _decodeReport(report);
+
+    (bool exist, uint32 roundId) = existReport(report_);
+    if (exist && msg.sender == s_secondaryProxy) {
```

Comment:
if you need to get the `roundId` and you are assuming the sender is pre-verified, this is where you should do the lookup given the `bytes` of the round itself.

maybe this is where you call `_decodeReport` and get the `roundId` with a new `internal` function.